### ISSUE TITLE
Deadlocks counter vs gauge data type

### ISSUE THREAD

**Issue by ionutzp** [wrouesnel/postgres_exporter#346]
Hello 
I was thinking deadlocks be represented by gauge data type in prometheus as it can increase and decrease. 
If a deadlock is released by postgres shouldn't the counter decrease it's value and then maybe a gauge is better suited for this?

```
"deadlocks":      {COUNTER, "Number of deadlocks detected in this database", nil, nil},
```

https://github.com/wrouesnel/postgres_exporter/blob/master/cmd/postgres_exporter/postgres_exporter.go#L242

**Comment by mbanck** [comment]
I believe this counter just counts the number of deadlocks that have ever (well, since the last `stats_reset`) occurred in the database, and does not track the current number of deadlocks. So I think having this as `COUNTER` is just as good as for all the other ones. 

I am not sure how a `stats_reset` would mess this, but apart from going back to zero when the statistics are reset (on every crash recovery, or when manually reset), the counter should go up monotonically which sounds right to me.

**Comment by keithf4** [comment]
Yes, this is just an incremental counter in postgres. It's better used in a formula that calculates the rate of change vs a static counter. This is the same for the table stats values that are gathered from `pg_stat_all_tables` and database level stats from `pg_stat_database`.

**Comment by sysadmind** [comment]
Thanks for the report! This issue has been inactive for a long time and there isn't enough detail to reproduce it against a current release. If this is still a problem with v0.20.0, please open a new issue with up-to-date details and we'll take a look. Closing as stale for now.
